src/utils/workspace.ts

```diff
@@ -23,12 +23,9 @@ export async function selectWorkspaceFolder(ui: IUserInterface, placeholder: str
 
 export function isFolderOpenInWorkspace(fsPath: string): boolean {
     if (vscode.workspace.workspaceFolders) {
-        if (!fsPath.endsWith(path.sep)) {
-            fsPath = fsPath + path.sep;
-        }
 
         const folder: vscode.WorkspaceFolder | undefined = vscode.workspace.workspaceFolders.find((f: vscode.WorkspaceFolder): boolean => {
-            return fsPath.startsWith(f.uri.fsPath);
+            return path.relative(fsPath, f.uri.fsPath) === '';
```

Comment:
My concern is if the subfolder is not opened, the user might not able to launch it since the _cwd_
 might not have ```.vscode``` folder which contains ```launch.json``` and ```task.json```. 

Another concern is that as what we discussed above. The maven archetype will always create a artifactId folder first, and put function things into the folder. For Java, there is another important file: ```pom.xml```, will also be put into the artifactId folder. The maven command can be execute only if the _cwd_ contains ```pom.xml``` or specify its location in parameter. That's why we prefer to open the folder which contains pom.xml (that is the artifactId folder).

I quite agree that always open the project base path might not be the best solution for this. While I think dong this it not that harmful. What about keeping this change for now and file a issue and we can do more further discussion in detail?

Reply:
Yeah filing an issue is fine

Reply:
Greate